### DISCUSSION TITLE
Determine download amount of an URL

### DISCUSSION THREAD

**Discussion by dukesaeed** [cefsharp/CefSharp#4926] [Q&A]
Hello Community

How can i determine download/upload amount of an URL?

ChatGPT offers something like this:

```
public class MyResourceHandler : DefaultResourceHandler
{
    private long downloadedData = 0;

    public override Stream GetResponse(IResponse response, out long responseLength, out string redirectUrl)
    {
        var stream = base.GetResponse(response, out responseLength, out redirectUrl);

        // Add the size of the response data to the downloadedData counter
        if (responseLength > 0)
        {
            downloadedData += responseLength;
        }

        return stream;
    }

    public long GetDownloadedDataSize()
    {
        return downloadedData;
    }
}
```

but obviousely this is not working 

Regards

**Comment by amaitland** [comment]
`ResourceRequestHandler.OnResourceLoadComplete` should give you the `receivedContentLength` on each resource.

See https://github.com/cefsharp/CefSharp/wiki/General-Usage#request-handling for a generic example of `ResourceRequestHandler`, you can override the methods you are interested in.
